src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2960,6 +2960,171 @@ mesh::topology::unstructured::generate_sides(const Node &topo,
     blueprint::o2mrelation::generate_offsets(d2smap, info);
 }
 
+//-----------------------------------------------------------------------------
+template<typename T, typename U> 
+void 
+map_field(Node &field_out, 
+          const Node &field, 
+          int num_shapes, 
+          const U *tri_to_poly)
+{
+    T* values_array = field_out["values"].value();
```

Comment:
one thing I forgot to mention -- there could be multi component fields here. I can discuss with you in the future.